src/pages/index.js

```diff
@@ -41,48 +51,110 @@ const enableValidation = (config) => {
 };
 enableValidation(config);
 
-const createCard = (cardObject) => {
-  const card = new Card(cardObject, "#card__template", () => {
-    imagePopup.open(cardObject);
+const handleImageClick = (data) => {
+  imagePopup.open(data);
+};
+
+const handleDelete = (card) => {
+  confirmationPopup.open();
+  confirmationPopup.setSubmitAction(() => {
+    api
+      .deleteCard(card)
+      .then(() => {})
+      .catch((err) => console.error(err));
+    confirmationPopup.close();
+    card.handleTrashBtn();
   });
+};
+
+const handleLike = (data, specificCard) => {
+  api
+    .likeCard(data)
+    .then((results) => {
+      specificCard.setLike(results.isLiked);
+    })
+    .catch((err) => console.error(err));
+};
+
+const createCard = (data) => {
+  const card = new Card(
+    data,
+    "#card__template",
+    handleImageClick,
+    handleDelete,
+    handleLike
+  );
   return card.generateCard();
 };
 
 const imagePopup = new PopupWithImage(".picture-modal");
 imagePopup.setEventListener();
 
-const cardSection = new Section(
-  {
-    data: initialCards,
-    renderer: (data) => {
-      const cardElement = createCard(data);
-      cardSection.addItem(cardElement);
-    },
-  },
-  cardsList
-);
-cardSection.renderItems();
-
 const newCardPopup = new PopupWithForm(".add-card-modal", (data) => {
-  const cardElement = createCard(data);
-  cardSection.addItem(cardElement);
+  newCardPopup.renderLoading(true);
   newCardPopup.close();
+  api
+    .addNewCard(data)
+    .then((results) => {
+      const cardElement = createCard(results);
+      cardSection.addItem(cardElement);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      newCardPopup.renderLoading(false);
+    });
 });
 newCardPopup.setEventListener();
 
-const userInfo = new UserInfo(".profile__name", ".profile__occupation");
+const userInfo = new UserInfo(
+  ".profile__name",
+  ".profile__occupation",
+  ".profile__avatar"
+);
 
 const profilePopup = new PopupWithForm(".profile-modal", (data) => {
+  profilePopup.renderLoading(true);
   profilePopup.close();
-  userInfo.setUserInfo(data);
+  api
+    .editUserInfo(data)
+    .then((results) => {
+      userInfo.setUserInfo(results);
+    })
+    .catch((err) => console.error(err))
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors